### PR TITLE
Adding unit testing coverage for functions `wm_sync_group_file` and `wm_sync_legacy_groups_files`

### PR DESCRIPTION
**Pull request by DProvinciani**
|Related issue|
|---|
| https://github.com/wazuh/wazuh/issues/11754 |

This pull request adds unit testing coverage for the functions `wm_sync_group_file` and `wm_sync_legacy_groups_files` of the `wm_database.c` synchronization module.

In addition, it modifies some logs that were not using the tag of the `wm_database.c` synchronization module.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade

- Memory tests for Linux
  - [x] Scan-build report
image

  - [x] AddressSanitizer

- [x] Added unit tests (for new features)
image
image

### CODE REVIEW

File: src/unit_tests/wazuh_modules/database/test_wm_database.c [hunk 1]
```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright (C) 2015, Wazuh Inc.
```
Comment (san1526) [inline src/unit_tests/wazuh_modules/database/test_wm_database.c:2]: minor, `Copyright (C) 2022`

**Review by san1526 — APPROVED** [review]
LGTM

File: src/unit_tests/wazuh_modules/database/test_wm_database.c [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+ */
+char *generate_groups_csv_string(unsigned int ngroups)
+{
+    char *groups = NULL;
+    os_calloc(OS_BUFFER_SIZE, sizeof(char), groups);
+
+    for (int i = 0, position = 0; i < ngroups + 1; ++i) {
+        snprintf(groups+position, OS_BUFFER_SIZE - position, (ngroups-1 == i) ? "group%d\n" : "group%d,", i);
+        position = strlen(groups);
+    }
```
[the remark this answers is not in the document]
Comment (DProvinciani) [reply src/unit_tests/wazuh_modules/database/test_wm_database.c:65]: Fixed

File: src/unit_tests/wazuh_modules/database/test_wm_database.c (hunk 1 above)
Comment (DProvinciani) [reply src/unit_tests/wazuh_modules/database/test_wm_database.c:2]: Fixed

**Pull request merged**